teamcity: failed tests on master: acceptance/TestGossipRestart

The following tests appear to have failed:

[#315406](https://teamcity.cockroachdb.com/viewLog.html?buildId=315406):

```
--- FAIL: acceptance/TestGossipRestart (23.070s)
test_log_scope.go:78: test logs captured to: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/logTestGossipRestart189308734
test_log_scope.go:61: use -show-logs to present logs inline
localcluster.go:727: expected event &{2 restart}, got &{0 restart} (after [{0 die} {1 die} {2 die} {3 die} {1 restart} {3 restart}])
------- Stdout: -------
xenial-20170214: Pulling from library/ubuntu
Digest: sha256:dd7808d8792c9841d0b460122f1acf0a2dd1f56404f8d1e56298048885e45535
Status: Image is up to date for ubuntu:xenial-20170214
xenial-20170214: Pulling from library/ubuntu
Digest: sha256:dd7808d8792c9841d0b460122f1acf0a2dd1f56404f8d1e56298048885e45535
Status: Image is up to date for ubuntu:xenial-20170214
test logs left over in: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/logTestGossipRestart189308734

```
Please assign, take a look and update the issue accordingly.

--- a/pkg/acceptance/gossip_peerings_test.go
+++ b/pkg/acceptance/gossip_peerings_test.go
@@ -128,13 +128,24 @@ func testGossipRestartInner(
 		}
 
 		log.Infof(ctx, "restarting all nodes")
-		ch := make(chan error)
-		for i := 0; i < num; i++ {
-			go func(i int) { ch <- c.Restart(ctx, i) }(i)
-		}
-		for i := 0; i < num; i++ {
-			if err := <-ch; err != nil {
-				t.Errorf("error restarting node %d: %s", i, err)
+		if _, ok := c.(*cluster.LocalCluster); ok {
+			// It is not safe to call Restart in parallel when using
+			// cluster.LocalCluster because expected container shutdown events
+			// may be reordered.
+			for i := 0; i < num; i++ {
+				if err := c.Restart(ctx, i); err != nil {
+					t.Errorf("error restarting node %d: %s", i, err)
+				}
+			}
+		} else {
+			ch := make(chan error)
+			for i := 0; i < num; i++ {
+				go func(i int) { ch <- c.Restart(ctx, i) }(i)
+			}
+			for i := 0; i < num; i++ {
+				if err := <-ch; err != nil {
+					t.Errorf("error restarting node %d: %s", i, err)
+				}
 			}
 		}
 		if t.Failed() {

